### PR TITLE
Fix Clang-15 warnings

### PR DESCRIPTION
**Pull request by AtariDreams**
Redundant casts, noexcept for move constructors, and usage of default and delete make the code more performant and maintainable

### CODE REVIEW

**Comment by ckormanyos** [comment]
Hi @AtariDreams I took a quick glance at the drone run in CI (which runs even without admin approval).

To my rather quick glance it seems like a lot of tests fail with these changes. I'm not sure exactly what went wrong, but this should be investigated prior to running all of CI.

Cc: @jzmaddock and @NAThompson and @mborland

**Comment by AtariDreams** [comment]
Fixed!

**Comment by ckormanyos** [comment]
Hi @AtariDreams thanks for correcting these suggested corrections.

Could we wait a few hours or half a day or so before approving CI? John (@jzmaddock) has been working steadily and dilligently on some special functions correctins and these are cacling now.

I (or somebody) will approve this run after some of heavy action finishes up, which should be in a few hours.

Cc: @jzmaddock and @mborland and @NAThompson

**Comment by ckormanyos** [comment]
To be more clear @AtariDreams to me it would seem nice to get #835 (cycling in CI now) merged (see work by @jzmaddock). Then merge that updated develop into your branch and then run all of CI.

Thoughts?

**Comment by AtariDreams** [comment]
That makes sense. Perfect!

File: include/boost/math/tools/polynomial_gcd.hpp [hunk 1]
```diff
@@ -23,7 +23,6 @@
 #include <numeric>
 #include <utility>
 #include <iterator>
-#include <type_traits>
```
Comment (jzmaddock) [inline include/boost/math/tools/polynomial_gcd.hpp:26]: This header use std::is_arithmetic: surely we should keep this #include?

Comment (jzmaddock) [reply include/boost/math/tools/polynomial_gcd.hpp:26]: Ah no I see, yes you're quite right!

**Pull request merged**